### DISCUSSION TITLE
Apparent TypeScript type issue with Vitest test extends

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [vitest-dev/vitest#5710] [Q&A] [answered]
![CleanShot 2024-05-12 at 09 35 32@2x](https://github.com/vitest-dev/vitest/assets/284476/bdf24d4a-8d6f-4a2d-b734-f00c183026ff)

Can someone confirm this is a Vitest issue and not me? I'll open up an issue if it is Vitest. As best I can tell, the typing issue stems from a Vitest issue and not user error here. My TSConfig etc. is not doing anything special, strict mode etc. modern setup.

**Comment by hi-ogawa** [comment, marked as answer]
That should intend to work, so it could be some edge case bug where type inference is confused by `Mock` type or tsconfig.

It looks working on simple test on stackblitz https://stackblitz.com/edit/vitest-dev-vitest-w4tsmc?file=test%2Frepro.test.ts, so we'll need a concrete reproduction. Feel free to open an issue if you have one.

**Reply by jasonkuhrt** [reply]
This issue is resolved in Vitest 2.
